game-core/src/main/java/games/strategy/triplea/image/UnitImageFactory.java

```diff
@@ -142,20 +148,55 @@ public Image getImage(final UnitCategory unit) {
       final String baseImageName, final PlayerId id, final ResourceLoader resourceLoader) {
     // URL uses '/' not '\'
     final String fileName = FILE_NAME_BASE + id.getName() + "/" + baseImageName + ".png";
-    final URL url = resourceLoader.getResource(fileName);
+    final String fileName2 = FILE_NAME_BASE + baseImageName + ".png";
+    final URL url = resourceLoader.getResource(fileName, fileName2);
     return Optional.ofNullable(url);
   }
 
-  private Optional<Image> getBaseImage(final String baseImageName, final PlayerId id) {
+  private Optional<Image> getBaseImage(
+      final String baseImageName, final PlayerId id, final UnitType type) {
     final Optional<URL> imageLocation = getBaseImageUrl(baseImageName, id);
     Image image = null;
     if (imageLocation.isPresent()) {
       image = Toolkit.getDefaultToolkit().getImage(getBaseImageUrl(baseImageName, id).get());
       Util.ensureImageLoaded(image);
+      if (!mapData.ignoreColorizingUnit(type.getName())) {
+        if (mapData.getUnitColor(id.getName()).isPresent()) {
+          final Color color = mapData.getUnitColor(id.getName()).get();
+          image = convertToBufferedImage(image);
+          ImageColorizer.applyColor(color, (BufferedImage) image);
+          if (mapData.shouldFlipUnit(id.getName())) {
+            image = flipImage((BufferedImage) image);
+          }
+        } else if (mapData.getUnitHsb(id.getName()).isPresent()) {
+          final int[] hsb = mapData.getUnitHsb(id.getName()).get();
```

Comment:
Nit: The if/else block contains some redundant parts.

```
image = convertToBufferedImage(image);
 
if (mapData.getUnitColor(id.getName()).isPresent()) {
  ImageColorizer.applyColor(color, (BufferedImage) image);
 } else if (mapData.getUnitHsb(id.getName()).isPresent()) {
    final int[] hsb = mapData.getUnitHsb(id.getName()).get();
    ImageColorizer.applyHsb(hsb[0], hsb[1], hsb[2], (BufferedImage) image);
 }

if (mapData.shouldFlipUnit(id.getName())) {
  image = flipImage((BufferedImage) image);
 }
```

Further though, the branching logic here mirrors a bit what is done in the colorizer class.

Notably:
```
    final float[] hsb = Color.RGBtoHSB(color.getRed(), color.getGreen(), color.getBlue(), null);
    applyHsb((int) (hsb[0] * 360), (int) (hsb[1] * 100), 0, image);
```

I think then if the map data method returned a normalized HSB directly, then there would be no need for branching logic here and everything would be just HSB.

It is a bit nasty for map data to do that computation, I've also been considering if it really should know as much about value parsing as it does as well. To that extent, I'd recommend to create a strategy object that takes a mapData object, eg:

```
class UnitColorParser {
  private final MapData mapData;
   public Optional<int[]> getUnitHsb(PlayerId player) {
            // read map Data property
            // do the string splitting
            // if it is a color value, convert to HSB
            // return hsb value
   }
}
```
The test case for such a class would be pretty straight forward and help ensure we got all the error cases correct. We probably should reject extreme and out of bound values.

Reply:
Refactored this and updated to handle brightness instead of HSB property. Its definitely a step in the right direction I think.